docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,727 @@
+
+## Designing and Preparing
+
+### Learning 
+  1. You should be familiar with using Kubernetes already.  We suggest you set
+    up a temporary cluster by following one of the other Getting Started Guides.
+    This will help you become familiar with the CLI ([kubectl](../kubectl.md)) and concepts ([pods](../pods.md), [services](../services.md), etc.) first.
+  1. You should have `kubectl` installed on your desktop.  This will happen as a side
+    effect of completing one of the other Getting Started Guides.
+
+### Cloud Provider
+Kubernetes has the concept of a Cloud Provider, which is a module which provides
+an interface for managing TCP Load Balancers, Nodes (Intances) and Newtworking Routes.
+The interface is defined in `pkg/cloudprovider/cloud.go`.  It is possible to
+create a custom cluster without implementing a cloud provider (for example if using
+bare-metal), and not all parts of the interface need to be implemented, depending
+on how flags are set on various components. 
+
+### Nodes
+- You can use virtual or physical machines.
+- The machines need to be on the same subnet.
+- While you can build a cluster with 1 machine, in order to run all the examples and tests you
+  need at least 4 nodes.
+- Many Getting-started-guides make a distinction between the master node and regular nodes.  This
+  is not strictly necessary.
+- Nodes will need to run some version of Linux with the x86_64 architecture.  It may be possible
+  to run on other OSes and Architectures, but this guide does not try to assist with that.
+- Apiserver and etcd together are fine on a machine with 1 core and 1GB RAM for clusters with 10s of nodes.
+  Larger or more active clusters may benefit from a more cores.
+- Other nodes can have any reasonable amount of memory and any number of cores.  They need not
+  have identical configurations.
+
+### Network
+Kubernetes has a distinctive [networking model](../networking.md).
+
+Kubernetes allocates an IP address to each pod and creates a virtual ethernet device for each
+Pod.  When creating a cluster, you need to allocate a block of IPs that Kubernetes to be used
+as Pod IPs.  The normal approach is to allocate a different block to each node in the cluster
+as the node is added.  A process in one pod should be able to communicate with another pod
+using the IP of the second pod.  This connectivity can be accomplished in two ways:
+- Configure network to route Pod IPs
+  - Harder to setup from scratch.
+  - The Google Compute Engine ([GCE](gce.md)) and [AWS](aws.md) guides use this approach.
+  - Need to make the Pod IPs routable by programming routers, switches, etc.
+  - Can be configured external to kubernetes, or can implement in the "Routes" interface of a Cloud Provider module.
+- Create an Overlay network
+  - Easier to setup
+  - Traffic is encapsulated, so per-pod IPs are routable.
+  - Examples:
+    - [Flannel](https://github.com/coreos/flannel) 
+    - [Weave](http://weave.works/)
```

Comment:
might want to mention ovs, which is used by vagrant and openshift guys I believe. It is by far the most mature option compared to flannel and weave.